llvm/include/llvm/Support/SpecialCaseList.h

```diff
@@ -122,7 +122,6 @@ class SpecialCaseList {
     // Returns zero if no match is found.
     unsigned match(StringRef Query) const;
 
-  private:
     StringMap<std::pair<GlobPattern, unsigned>> Globs;
```

Comment:
Is this intended or can it be kept private?

Reply:
this is intended, as we don't want to just check a match, but rather find the longest match. hence we implement our own matching logic based on the globs. I didn't want to change behavior for existing callers as this is likely more expensive than what they're doing today. in addition to that we also need to post-process sections based on their line numbers to map diagnostic groups to individual diagnostics.

putting all of this into SpecialCaseList felt like an overkill, and since this is already an internal class to SpecialCaseList and inheritors, I just made the data non-private instead. If there's interest I can lift them from WarningsSpecialCaseList implementation to the base instead.

Reply:
This can be made protected at least though, right?

Reply:
The only places I see it used are within `class WarningsSpecialCaseList : public llvm::SpecialCaseList` -- what am I missing?

Reply:
Oh, I missed the inner class context, that makes sense, thanks! It unfortunate that we're losing encapsulation with the change, but I can hold my nose.

Reply:
thanks, and I'd like to address these in a follow-up refactoring, sent out https://discourse.llvm.org/t/refactoring-llvm-specialcaselist/82931 to see what people think about this.